lib/utils/buildCommandArgs.js

```diff
@@ -56,7 +56,7 @@ export default function buildCommandArgs(
     });
 
   // Use minimist to parse the args, and then build varidiac args and options.
-  const parsedArgs = parseArgs(passedArgs, types);
+  const parsedArgs = parseArgs(passedArgs, types, command._disableTypeCasting === true);
```

Comment:
Wouldn't it make more sense to remove the type casting option from this `types` object? (`types` are options passed to minimist).